### DISCUSSION TITLE
Documentation for the LP10 and LP100 math functions

### DISCUSSION THREAD

**Discussion by jsiddall** [OpenHantek/OpenHantek6022#333] [Ideas]
I had a look at the code and it appears the LP10 and LP100 math functions are single pole low pass filters using 0.1% (10/10000) or 1% (100/10000) of the on-screen display as the time constant of the filter.  If I understand the theory correctly, this works out to a cutoff frequency of:

10000 / (DeltaT * DecayT * 2 * pi)

Where DeltaT is the amount of time displayed on the screen (in seconds) and DecayT is the decay time of the filter (in samples, 10 or 100 samples depending on whether LP10 or LP100 was selected).

For example, with a timebase setting of 200 us (i.e. DeltaT is 0.002 s on screen) a LP100 filter is effectively a zero phase low pass filter down 3 dB at 7.96 kHz.

Does that sound right?

**Comment by Ho-Ro** [comment]
You're right, these are zero phase (non-causal forward-backward) filters. I didn't check your numbers, but I'm sure you calculated correctly.
I use this scope program also an experimental platform to try out some ideas - also the sinc interpolation and the musical note detection was this proof of concept.

**Reply by jsiddall** [reply]
Yes, the LP filters are nice things to have.  Could some info on the low pass filters be added to the next version of the user guide?

BTW: I also like sinc interpolation, and use it as my my preferred setting!
